va/va.go

```diff
@@ -664,12 +664,7 @@ type validationLogEvent struct {
 // implements the DCV portion of Multi-Perspective Issuance Corroboration as
 // defined in BRs Sections 3.2.2.9 and 5.4.1.
 func (va *ValidationAuthorityImpl) DoDCV(ctx context.Context, req *vapb.PerformValidationRequest) (*vapb.ValidationResult, error) {
-	// TODO(#8023): Once DnsNames are no longer used in RPCs, use req.Identifier
-	// directly instead of setting ident.
-	if req.Identifier != nil && req.DnsName != "" {
-		return nil, errors.New("both Identifier and DNSName are set")
-	}
-	ident := identifier.FromProtoWithDefault(req)
+	ident := identifier.FromProto(req.Identifier)
```

Comment:
Move this below core.IsAnyNilOrZero, and replace the check on `ident` there with a check on `req.Identifier`